Minor fixes and improvements

sjaeckel opened the pull request:
Title says all

e.g. update to latest ltm API

@jbech-linaro do you probably have the possibility to check/time to review that the "check ranges of more ciphers" commit does indeed what you did in #493 for AES?

sjaeckel committed:
clean-up define

sjaeckel committed:
check ranges of more ciphers

Inspired by #493 of @jbech-linaro I reviewed all blockciphers
for similar patterns.

Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

sjaeckel committed:
Fix includes

Headers which are included with '< >' can only be found if the folder where
the files are located is given as include path.
To be able to install the files to a separate folder, include them instead
with '" "'.

sjaeckel committed:
run {A,UB}SAN again with GMP

Fixup of 798d7dc61f0cad04972ddebd09c98136fbacb806

sjaeckel committed:
further updated ltm API

sjaeckel committed:
update LTC_DEPRECATED macro from ltm

sjaeckel committed:
use mp_init_copy() instead of init()+copy()

sjaeckel committed:
don't produce any console-output on `make V=0`

jockebech commented on src/ciphers/rc5.c:
```diff
@@ -136,6 +136,10 @@ int rc5_ecb_encrypt(const unsigned char *pt, unsigned char *ct, const symmetric_
    LTC_ARGCHK(pt   != NULL);
    LTC_ARGCHK(ct   != NULL);
 
+   if (skey->rc5.rounds < 12 || skey->rc5.rounds > 24) {
```
I'm not too familiar with the inner workings of RC5, but the lower limit seems fine, however for the upper limit I'm not sure what the correct value would be (max value according to spec is 255 IIRC). But it's probably unlikely that one would use a greater value than 24. Related to this, `rfc2040` seems to define `MAX_ROUNDS` to 20.

sjaeckel replied:
thx!

> Related to this, `rfc2040` seems to define `MAX_ROUNDS` to 20

that value is implementation defined as is 24 for ltc.

jockebech reviewed (commented):
For commit b96e137b8f644f6642322f0c63c829a6f0a6e4cf 
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

sjaeckel commented:
I just asked myself if
1. it'd make sense to review the default rounds of all the ciphers? as e.g. as of [1] our default value of 12 rounds for RC5 can be attacked
2. we can even change those values? as this will break users who rely on 0 being that then-old default number of rounds...

[1] https://en.wikipedia.org/w/index.php?title=RC5&oldid=917621181#cite_note-Biryukov-1

sjaeckel committed:
re-order vars of travis.yml

the travis UI shows them right besides the build and the name says more
than the script on what is done

sjaeckel committed:
add testcase for issue #507

sjaeckel committed:
re-order examples

sjaeckel committed:
fix #469

[skip ci]
